portal-impl/src/com/liferay/portlet/PortalPreferencesImpl.java

```diff
@@ -495,30 +443,25 @@ protected String toXML() {
 		Element portletPreferencesElement = new Element(
 			"portlet-preferences", false);
 
-		for (Map.Entry<String, String[]> entry : preferences.entrySet()) {
-			String[] values = entry.getValue();
+		for (Map.Entry<PortalPreferenceKey, String[]> entry :
+				preferences.entrySet()) {
+
+			PortalPreferenceKey portalPreferenceKey = entry.getKey();
 
 			Element preferenceElement = portletPreferencesElement.addElement(
 				"preference");
 
-			preferenceElement.addElement("name", entry.getKey());
+			preferenceElement.addElement(
+				"name", portalPreferenceKey.toString());
```

Comment:
@shuyangzhou this is where `PortalPreferenceKey.toString()` is used.  Maybe we give it a different method like `getName()`?